src/osd/OSD.h

```diff
@@ -1421,8 +1421,7 @@ class OSD : public Dispatcher,
 
     Spinlock sent_epoch_lock;
     epoch_t last_sent_epoch;
-    Spinlock received_map_lock;
-    epoch_t received_map_epoch; // largest epoch seen in MOSDMap from here
+    std::atomic<epoch_t> received_map_epoch; // largest epoch seen in MOSDMap from here
```

Comment:
There are definitely compare-and-swap atomics in the hardware; I *presume* C++ supports them... If not I will be a sad panda. :(